docs/development.md

```diff
@@ -26,6 +26,7 @@
       - [Customizing the cluster deployment](#customizing-the-cluster-deployment)
       - [Creating the cluster](#creating-the-cluster)
     - [Debugging cluster creation](#debugging-cluster-creation)
+    - [Debugging workload cluster creation](#debugging-workload-cluster-creation)
```

Comment:
these two debugging sections should be combined into one section (they're both about debugging workload cluster creation), or even a separate doc as discussed in #520. Slight preference for a separate doc because debugging cluster creation steps don't only apply to development, they also apply to users.